docs/class_diagram.uml

```diff
@@ -0,0 +1,438 @@
+        + WriteNotificationRecordImpl(orgID types.OrgID, accountNumber types.AccountNumber, clusterName types.ClusterName, notificationTypeID types.NotificationTypeID, stateID types.StateID, report types.ClusterReport, updatedAt types.Timestamp, notifiedAt types.Timestamp, errorLog string) error
+
+    }
+}
+
+"main.Storage" <|-- "main.DBStorage"
+
+namespace mocks {
+    class Producer << (S,Aquamarine) >> {
+        + Close() error
+        + New(brokerCfg conf.KafkaConfiguration) (*producer.KafkaProducer, error)
+        + ProduceMessage(msg types.NotificationMessage) (int32, int64, error)
+
+    }
+    class Storage << (S,Aquamarine) >> {
+        + Close() error
+        + ReadClusterList() ([]types.ClusterEntry, error)
+        + ReadLastNNotificationRecords(clusterEntry types.ClusterEntry, numberOfRecords int) ([]types.NotificationRecord, error)
+        + ReadNotificationTypes() ([]types.NotificationType, error)
+        + ReadReportForCluster(orgID types.OrgID, clusterName types.ClusterName) (types.ClusterReport, types.Timestamp, error)
+        + ReadReportForClusterAtOffset(orgID types.OrgID, clusterName types.ClusterName, offset types.KafkaOffset) (types.ClusterReport, error)
+        + ReadReportForClusterAtTime(orgID types.OrgID, clusterName types.ClusterName, updatedAt types.Timestamp) (types.ClusterReport, error)
+        + ReadStates() ([]types.State, error)
+        + WriteNotificationRecord(notificationRecord types.NotificationRecord) error
+        + WriteNotificationRecordForCluster(clusterEntry types.ClusterEntry, notificationTypeID types.NotificationTypeID, stateID types.StateID, report types.ClusterReport, notifiedAt types.Timestamp, errorLog string) error
+        + WriteNotificationRecordImpl(orgID types.OrgID, accountNumber types.AccountNumber, clusterName types.ClusterName, notificationTypeID types.NotificationTypeID, stateID types.StateID, report types.ClusterReport, updatedAt types.Timestamp, notifiedAt types.Timestamp, errorLog string) error
+        + CleanupForOrganization(orgID types.OrgID, maxAge string, statement string) (int, error)
+        + CleanupNewReportsForOrganization(orgID types.OrgID, maxAge string) (int, error)
+        + CleanupOldReportsForOrganization(orgID types.OrgID, maxAge string) (int, error)
+
+    }
+}
+"mock.Mock" *-- "mocks.Producer"
+"mock.Mock" *-- "mocks.Storage"
+
+"differ.Storage" <|-- "mocks.Storage"
+"main.Storage" <|-- "mocks.Storage"
+
+namespace producer {
+    class KafkaProducer << (S,Aquamarine) >> {
+        + Configuration conf.KafkaConfiguration
+        + Producer sarama.SyncProducer
+
+        + ProduceMessage(msg types.NotificationMessage) (int32, int64, error)
+        + Close() error
+
+    }
+    interface Producer  {
+        + New(brokerCfg conf.KafkaConfiguration) (Producer, error)
+        + ProduceMessage(msg types.NotificationMessage) (int32, int64, error)
+        + Close() error
+
+    }
+}
+
```

Comment:
missing a "producer.Producer" <|-- "producer.KafkaProducer" relation

Reply:
dunno why as it is autogenerated ;)

Reply:
Ah! Maybe something is wrong with my KafkaProducer then ^^